app/my-bets/page.tsx

```diff
@@ -7,96 +7,119 @@ import { Market, Position, tradesOutcomeBalance } from '@/entities';
-          const isWinner = market.isWinner(outcomeIndex);
+  const filterActiveBets = useMemo(
+    () =>
+      userPositionsComplete?.filter(
+        userPosition => !userPosition.position.conditions[0].resolved
+      ) ?? [],
+    [userPositionsComplete]
+  );
 
-          const isResolved = userPositionCondition.resolved;
-          const hasPayoutDenominator = +userPositionCondition.payoutDenominator > 0;
+  const filterCompleteBets = useMemo(
+    () =>
+      userPositionsComplete?.filter(
+        userPosition => userPosition.position.conditions[0].resolved
+      ) ?? [],
+    [userPositionsComplete]
+  );
 
-          const canClaim = isWinner && isResolved && !isClaimed && hasPayoutDenominator;
+  const filterUnredeemedBets = useMemo(() => {
+    return filterCompleteBets.filter(userPosition => {
+      const position = new Position(userPosition.position);
+      const outcomeIndex = position.outcomeIndex - 1;
 
-          if (canClaim) return userPosition;
-        } catch (error) {
-          console.error(error);
-        }
-      })
-    );
+      const condition = userPosition.condition;
+      const market = condition && new Market(condition?.fixedProductMarketMakers[0]);
+      const outcomeBalance = tradesOutcomeBalance({
+        fpmmTrades: userPosition.fpmmTrades,
+      });
+      const userPositionCondition = userPosition.position.conditions[0];
 
-    const filteredResults: UserPosition[] = results.filter(
-      (result): result is UserPosition => result !== undefined
-    );
-    setUnredeemedBets(filteredResults);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [userPositions]);
+      const isClaimed = !outcomeBalance;
+      const isWinner = market.isWinner(outcomeIndex);
 
-  useEffect(() => {
-    if (!address || !filterCompleteBets.length) return;
+      const isResolved = userPositionCondition.resolved;
+      const hasPayoutDenominator = +userPositionCondition.payoutDenominator > 0;
 
-    fetchUnredeemedBets();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [userPositions]);
+      return isWinner && isResolved && !isClaimed && hasPayoutDenominator;
+    });
+  }, [filterCompleteBets]);
 
   if (!address) return <NoWalletConnectedPage />;
-  if (data && data.userPositions.length === 0) return <NoBetsPage />;
+  if (userPositionsComplete && userPositionsComplete.length === 0) return <NoBetsPage />;
+
+  const sortBetsByNewest = (bets: UserPositionComplete[]) => {
+    return [...bets].sort((a, b) => {
+      return (
+        b.fpmmTrades[b.fpmmTrades.length - 1]?.creationTimestamp -
+        a.fpmmTrades[a.fpmmTrades.length - 1]?.creationTimestamp
+      );
+    });
+  };
+
+  const sortedAllBets = sortBetsByNewest(userPositionsComplete ?? []);
+  const sortedCompleteBets = sortBetsByNewest(filterCompleteBets);
+  const sortedActiveBets = sortBetsByNewest(filterActiveBets);
+  const sortedUnredeemedBets = sortBetsByNewest(filterUnredeemedBets);
```

Comment:
Let's sort before filtering. So we first sort, then we filter by complete, active, unredeemed.

Reply:
yes